ts/screens/services/ServicesHomeScreen.tsx

```diff
@@ -88,22 +135,84 @@ const styles = StyleSheet.create({
   },
   organizationLogo: {
     marginBottom: 0
+  },
+  center: {
+    alignItems: "center"
+  },
+  padded: {
+    paddingHorizontal: customVariables.contentPadding
+  },
+  customSpacer: {
+    height: customVariables.spacerHeight + customVariables.h1LineHeight
+  },
+  errorText: {
+    fontSize: customVariables.fontSize2,
+    paddingTop: customVariables.contentPadding
+  },
+  errorText2: {
+    fontSize: customVariables.fontSizeSmall
+  },
+  buttonBar: {
+    flexDirection: "row",
+    zIndex: 1,
+    justifyContent: "space-around",
+    backgroundColor: customVariables.colorWhite,
+    padding: 10
+  },
+  buttonBarLeft: {
+    flex: 2,
+    marginEnd: 5
+  },
+  buttonBarRight: {
+    flex: 2,
+    marginStart: 5
   }
 });
 
 const AnimatedTabs = Animated.createAnimatedComponent(Tabs);
-/**
- * A screen that contains all the Tabs related to services.
- */
+
 class ServicesHomeScreen extends React.Component<Props, State> {
   private navListener?: NavigationEventSubscription;
+
   constructor(props: Props) {
     super(props);
     this.state = {
-      currentTab: 0
+      currentTab: 0,
+      currentTabServicesId: [],
+      enableHeaderAnimation: false,
+      isLongPressEnabled: false,
+      enableServices: false
     };
   }
 
+  private updateLongPressButtonScope = () => {
+    const currentTabServicesChannels = this.props.getServicesChannels(
+      this.props.tabsServicesId[this.state.currentTab],
+      this.props.profile
+    );
+
+    const disabledServices: number = Object.keys(
+      currentTabServicesChannels
+    ).filter(id => currentTabServicesChannels[id].indexOf("INBOX") !== -1)
+      .length;
+
+    if (disabledServices === Object.keys(currentTabServicesChannels).length) {
+      // if all tab services are disabled the footer primary button allows services are massively enabled
+      this.setState({ enableServices: true });
+    } else {
+      // if at least one tab service is enabled the footer primary button allows services are massively disabled
+      this.setState({ enableServices: false });
+    }
+  };
+
+  private handleOnLongPressItem = () => {
+    this.updateLongPressButtonScope();
+    this.setState({
```

Comment:
```suggestion
    const isLongPressEnabled = !this.state.isLongPressEnabled;
    const currentTabServicesId = this.props.tabsServicesId[
      this.state.currentTab
    ];
    this.setState({
      isLongPressEnabled,
      currentTabServicesId
    });
```
this is to prevent potentially inconsistent value while state is updating (reading a value that is changing)

Reply:
updated within https://github.com/teamdigitale/io-app/pull/1269/commits/fab3d9b2a98e2d08e181bc5713f484ca40754572